src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/ApprovalStep.tsx

```diff
@@ -197,3 +210,81 @@ export const ApprovalStep = ({
     />
   )
 }
+
+const ApprovalStepComplete = ({
+  tradeQuoteStep,
+  hopIndex,
+  isLastStep,
+  isLoading,
+}: ApprovalStepProps) => {
+  const {
+    state,
+    approval: { txHash, state: approvalTxState },
+  } = useAppSelector(state => selectHopExecutionMetadata(state, hopIndex))
+
+  const isError = useMemo(
+    () => approvalTxState === TransactionExecutionState.Failed,
+    [approvalTxState],
+  )
+
+  const stepIndicator = useMemo(() => {
+    return <ApprovalStatusIcon hopExecutionState={state} approvalTxState={approvalTxState} />
+  }, [approvalTxState, state])
+
+  const description = useMemo(() => {
+    return (
+      <ApprovalDescription
+        tradeQuoteStep={tradeQuoteStep}
+        isError={isError}
+        txHash={txHash}
+        approvalNetworkFeeCryptoFormatted={undefined}
+      />
+    )
+  }, [isError, tradeQuoteStep, txHash])
+
+  return (
+    <StepperStep
+      title='Token allowance approval'
+      description={description}
+      stepIndicator={stepIndicator}
+      content={undefined}
+      isLastStep={isLastStep}
+      isLoading={isLoading}
+      isError={approvalTxState === TransactionExecutionState.Failed}
+      isPending={approvalTxState === TransactionExecutionState.Pending}
+    />
+  )
+}
+
+export const ApprovalStep = ({
+  tradeQuoteStep,
+  hopIndex,
+  isActive,
+  isLastStep,
+  isLoading,
+}: ApprovalStepProps) => {
+  const { state } = useAppSelector(state => selectHopExecutionMetadata(state, hopIndex))
+
+  // separate component for completed states to simplify hook dismount
+  if (state === HopExecutionState.AwaitingSwap || state === HopExecutionState.Complete) {
+    return (
+      <ApprovalStepComplete
+        tradeQuoteStep={tradeQuoteStep}
+        hopIndex={hopIndex}
+        isActive={isActive}
+        isLastStep={isLastStep}
+        isLoading={isLoading}
+      />
+    )
+  }
+
+  return (
+    <ApprovalStepPending
+      tradeQuoteStep={tradeQuoteStep}
+      hopIndex={hopIndex}
+      isActive={isActive}
+      isLastStep={isLastStep}
+      isLoading={isLoading}
+    />
+  )
+}
```

Comment:
all the changes in this file were to prevent approvals spamming api requests. not strictly related to this PR but was driving me insane while testing so here it is.